server/server.go

```diff
@@ -131,7 +132,16 @@ func (l *allowinglistener) Accept() (net.Conn, error) {
 		return conn, err
 	}
 
-	ip := conn.RemoteAddr().(*net.TCPAddr).IP.String()
+	ip := ""
+	remoteAddr := conn.RemoteAddr()
+	switch addr := remoteAddr.(type) {
+	case *net.TCPAddr:
+		ip = addr.IP.String()
+	case *net.UDPAddr:
+		ip = addr.IP.String()
+	default:
+		log.Errorf("Remote addr %v is of unknown type %v, unable to determine IP", remoteAddr, reflect.TypeOf(remoteAddr))
```

Comment:
Can we skip calling `allow` here? Passing an empty `ip` has no benefit and may lead to unexpected behavior.